test/test_fcl_geometric_shapes.cpp

```diff
@@ -2752,3 +2752,177 @@ BOOST_AUTO_TEST_CASE(conecone)
 
 
 
+template<typename S1, typename S2>
+void testReversibleShapeIntersection(const S1& s1, const S2& s2, FCL_REAL distance)
+{
+  Transform3f tf1(Vec3f(-0.5 * distance, 0.0, 0.0));
+  Transform3f tf2(Vec3f(+0.5 * distance, 0.0, 0.0));
+
+  Vec3f contactA;
+  Vec3f contactB;
+  FCL_REAL depthA;
+  FCL_REAL depthB;
+  Vec3f normalA;
+  Vec3f normalB;
+
+  bool resA;
+  bool resB;
+
+  const double tol = 1e-6;
+
+  resA = solver1.shapeIntersect(s1, tf1, s2, tf2, &contactA, &depthA, &normalA);
+  resB = solver1.shapeIntersect(s2, tf2, s1, tf1, &contactB, &depthB, &normalB);
+
+  BOOST_CHECK(resA);
+  BOOST_CHECK(resB);
+  BOOST_CHECK(contactA.equal(contactB, tol));  // contact point should be same
+  BOOST_CHECK(normalA.equal(-normalB, tol));  // normal should be opposite
+  BOOST_CHECK_CLOSE(depthA, depthB, tol);  // penetration depth should be same
+
+  resA = solver2.shapeIntersect(s1, tf1, s2, tf2, &contactA, &depthA, &normalA);
+  resB = solver2.shapeIntersect(s2, tf2, s1, tf1, &contactB, &depthB, &normalB);
+
+  BOOST_CHECK(resA);
+  BOOST_CHECK(resB);
+  BOOST_CHECK(contactA.equal(contactB, tol));
+  BOOST_CHECK(normalA.equal(-normalB, tol));
+  BOOST_CHECK_CLOSE(depthA, depthB, tol);
+}
+
+BOOST_AUTO_TEST_CASE(reversibleShapeIntersection_allshapes)
+{
+  // This test check whether a shape intersection algorithm is called for the
+  // reverse case as well. For example, if FCL has sphere-capsule intersection
+  // algorithm, then this algorithm should be called for capsule-sphere case.
+
+  // Prepare all kinds of primitive shapes (7) -- box, sphere, capsule, cone, cylinder, plane, halfspace
+  Box box(10, 10, 10);
+  Sphere sphere(5);
+  Capsule capsule(5, 10);
+  Cone cone(5, 10);
+  Cylinder cylinder(5, 10);
+  Plane plane(Vec3f(), 0.0);
+  Halfspace halfspace(Vec3f(), 0.0);
+
+  // Use sufficiently short distance so that all the primitive shapes can intersect
+  FCL_REAL distance = 5.0;
+
+  // If new shape intersection algorithm is added for two distinct primitive
+  // shapes, uncommnet associated lines. For example, box-sphere intersection
+  // algorithm is added, then uncommnet box-sphere.
```

Comment:
`uncommnet` -> `uncomment` here and the line above. Maybe do a search for any other instances of the misspelling